pkg/apis/v1beta1/ec2nodeclass_status.go

```diff
@@ -63,4 +63,7 @@ type EC2NodeClassStatus struct {
 	// cluster under the AMI selectors.
 	// +optional
 	AMIs []AMI `json:"amis,omitempty"`
+	// InstanceProfile contains the resolved instance profile for the role
```

Comment:
Worth making as deprecated?

Reply:
This isn't deprecated. This is the resolved value into the status, not in the spec